src/stores/my-account.ts

```diff
@@ -5,20 +5,42 @@ import {
   loginWithSecretKey,
   Signer,
 } from '@/utils/account'
+import { getWeb3AuthClientId } from '@/utils/env/client'
+import { generateRandomName } from '@/utils/random-name'
 import { LocalStorage } from '@/utils/storage'
+import { SubstrateRPC } from '@/utils/substrate-rpc'
+import * as bottts from '@dicebear/bottts'
+import { createAvatar } from '@dicebear/core'
+import { CHAIN_NAMESPACES, SafeEventEmitterProvider } from '@web3auth/base'
+import { Web3Auth } from '@web3auth/modal'
 import { useAnalytics } from './analytics'
 import { create } from './utils'
 
+const clientId = getWeb3AuthClientId()
+
+type AuthenticatedUser = {
+  name: string
+  encodedSecretKey: string
+  address: string
+  profilePic: string
+  email?: string
+  username?: string
+}
+
 type State = {
-  isInitialized?: boolean
-  isInitializedAddress?: boolean
+  web3Auth: Web3Auth | null
+  authenticatedUser: AuthenticatedUser | null
+  provider: SafeEventEmitterProvider | null
+  authMethod: AUTHENTICATION_METHODS
```

Comment:
authMethod should be AuthMethods | undefined, because the initial should be undefined because you haven't logged in